usb/ehci/ehci-hub.c

```diff
@@ -227,19 +364,14 @@ int ehci_roothubReq(usb_transfer_t *t)
 			ret = ehci_clearPortFeature(hub, setup->wIndex, setup->wValue);
 			break;
 		case REQ_GET_DESCRIPTOR:
-			ret = ehci_getDesc(hub, t->buffer, t->size);
+			ret = ehci_getDesc(hub, setup->wValue >> 8, setup->wValue & 0xff,  t->buffer, t->size);
 			break;
 		case REQ_SET_CONFIGURATION:
+			ret = 0;
```

Comment:
ret is zero already (line 343)
```suggestion
		case REQ_SET_CONFIGURATION:
		case REQ_SET_ADDRESS:
```
or `default:` if we don't care about invalid input